R/generic.R

```diff
@@ -172,3 +175,37 @@ has_call <- function(x, name) {
   }
   FALSE
 }
+
+
+methods <- function(generic) {
+  methods_rec(generic@methods, character())
+}
+methods_rec <- function(x, signature) {
+  if (!is.environment(x)) {
+    return(x)
+  }
+
+  # Recursively collapse environments to a list
+  methods <- lapply(names(x), function(class) methods_rec(x[[class]], c(signature, class)))
+  unlist(methods, recursive = FALSE)
+}
+
+generic_add_method <- function(generic, signature, method) {
```

Comment:
It would be helpful for me if we could talk about this, because I don't quite understand the iterated dispatch bit